### PR TITLE
ENG-152: Generate QR Code

### PR DESCRIPTION
**Pull request by rfougy**
# Summary
- Implemented QR Code that is display only per Linear [ticket](https://linear.app/mun-e/issue/ENG-152/display-user-id-qr-code-modal-component-name-my-mun-e-id-display-only)
- Includes adding `"jsx": "react-native",` to tsconfig.json file to remove linting errors across multiple files.

[Link to Linear Ticket](https://linear.app/mun-e/issue/ENG-152/display-user-id-qr-code-modal-component-name-my-mun-e-id-display-only)
# Attention
- I will be consulting with Caleb on the absence of the small bar placed on the top of the QR Modal component. I believe that it is best not to include it, as including it would require implied dragging functionality, increasing developer time as a result.

### CODE REVIEW

File: src/components/profile/qr-code/ButtonGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+const styles = StyleSheet.create({
+  container: {
+    position: "relative",
+    display: "flex",
+    flexDirection: "row",
+    width: "100%",
+    padding: 10,
+    backgroundColor: "rgba(148, 152, 149, 0.14)",
+    borderRadius: 10,
```
Comment (HoseaCodes) [inline src/components/profile/qr-code/ButtonGroup.tsx:84]: Border Radius should be 22

File: src/components/profile/qr-code/ButtonGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  },
+  activeButton: {
+    position: "absolute",
+    width: "50%",
+    height: "100%",
+    top: "25%",
+    backgroundColor: "white",
+    paddingVertical: 10,
+    paddingHorizontal: 20,
+    borderRadius: 10,
```
Comment (HoseaCodes) [inline src/components/profile/qr-code/ButtonGroup.tsx:94]: Border radius should be 14

**Comment by HoseaCodes** [comment]
1. Looks like it is missing the line element above the header,
2. The modalView borderRadius looks off, it does not say in the Figma. I think 50 looks closer.
3. The entire modal should be #F2F7F3, and it looks to be white in the code.
4. If we cannot round the corner of the images we can use this image:
 
Frame 47583


Screenshot 2024-10-13 at 9 02 28 PM

**Comment by rfougy** [comment]
> 1. Looks like it is missing the line element above the header,
> 2. The modalView borderRadius looks off, it does not say in the Figma. I think 50 looks closer.
> 3. The entire modal should be #F2F7F3, and it looks to be white in the code.
> 4. If we cannot round the corner of the images we can use this image:
> 
> Frame 47583
> 
> Screenshot 2024-10-13 at 9 02 28 PM

1. Regarding the line, I've reached out Caleb regarding this, as including the line would imply to the user that the modal can be dragged, which as of now will take more time to develop. Given that this is POC, should I include anyhow?
4. I can round out the image, however this has proven to conflict with the functionality of scanning the code strangely enough. Given this, I think it would make most sense to opt into using the logo you provided, that way the functionality remains and the UI passes. What do you think of this approach?

**Comment by HoseaCodes** [comment]
1. Let's add the line but minus the functionality for now.
2. For the QRCode we can opt into using the logo.

**Review by HoseaCodes — APPROVED** [review]

**Review by HoseaCodes — APPROVED** [review]

**Pull request merged**